### ISSUE TITLE
no-spaced-func rule flagging non-functions

### ISSUE THREAD

**Issue by nzakas** [eslint/eslint#407]
The `no-spaced-func` rule is being overly aggressive in its interpretation of functions. For example, this code will trigger the warning:

``` js
(function(){

    if (foo) {
        bar();
    }
}());
```

It will flag the `if` statement as part of the function because the rules looks at all of the token representing the `CallExpression` rather than just the first set of parens.

@mduvall - can you take a look at this?
